Assignments/Program.cs

```diff
@@ -1 +1,116 @@
+                     output += ones[10 + d3] + " ";
+                     break;
+                  }
+               case (State.B, 0 or (>= 2 and <= 9)): {
+                     s = State.C;
+                     if (n != 0) output += tens[n] + ((i == 2 && d3 > 0) ? "-" : " ");
+                     break;
+                  }
+               case (State.C, >= 0 and <= 9): {
+                     s = State.D;
+                     if (n != 0) output += ones[n] + " ";
+                     break;
+                  }
+               default: { s = State.D; break; }
+            }
+         }
+         if (part > 0 && i != 2) {
+            var (cond, key) = (i == 0) ? (num / 1000 % 1000 > 0 || num % 1000 > 0, "Million") : (num % 1000 > 0, "Thousand");
+            output += key + (cond ? ", " : " ");
+         }
+      }
+      return output;
+   }
+   enum State { A, B, C, D };
+   #region State Machine Diagram for converting any 3-digit Number to Words:
+   /*
+    *                                  if (n == 0) -->                          if (n == 0) --->                         if (n == 0) --->
+    *                             ┌------------>----------┐                ┌----------->-----------┐                ┌----------->-----------┐
+    *                             ↑                       ↓                ↑                       ↓                ↑                       ↓
+    *                             │    else --------->    │                │    else ---------->   │                │    else ---------->   │
+    *  o---->----A(n = 0...9)-->--⬛--->---[action1]--->---B(n = 0...9)-->--⬛--->---[action3]--->---C(n = 0...9)-->--⬛--->---[action4]--->---⬛-----D       
+    *                                                                      │                                                                │    
+    *                                                                      ↓     if (n == 1) -->                                            ↑
+    *                                                                      └----------->---------------[action2]---------------->-----------┘
+    *  
+    *  Where,
+    *    action1: output += (one/two/.../eight/nine) + "Hundred and"
+    *    action2: output += (eleven/twelve/.../eighteen/nineteen)
+    *    action3: output += (ten/eleven/.../eighty/ninety)
+    *    action4: output += (one/two/.../eight/nine)
+    */
+   #endregion
+
+   /// <summary>Converts input number to Roman Numerals</summary>
+   public static string ToRomans (int num) {
+      string output = "";
+      var romans = new char[7] { 'M', 'D', 'C', 'L', 'X', 'V', 'I' };
+      for (int i = 0, r = 1000; i < 7; i++) {
+         output += new string (romans[i], num / r);
+         num %= r;
+         r /= (i % 2 == 0) ? 2 : 5;
+      }
+      return output.Replace ("VIIII", "IX").Replace ("IIII", "IV").Replace ("LXXXX", "XC").Replace ("XXXX", "XL").Replace ("DCCCC", "CM").Replace ("CCCC", "CD");

```

Comment:
Looks complicated. Can you simplify this?
While adding to output itself you could replace right?